python/mujoco/usd/objects.py

```diff
@@ -334,42 +336,56 @@ def __init__(
         [3 for _ in range(mesh_facenum)]
     )
     self.usd_mesh.GetFaceVertexIndicesAttr().Set(mesh_face)
+    self._set_refinement_properties(self.usd_prim)
 
-    # setting mesh uv properties
-    mesh_texcoord, _ = self._get_uv_geometry()
-    self.texcoords = UsdGeom.PrimvarsAPI(self.usd_mesh).CreatePrimvar(
-        "UVMap", Sdf.ValueTypeNames.TexCoord2fArray, UsdGeom.Tokens.faceVarying
-    )
+    if self.geom_textures and self.geom_textures[mujoco.mjtTextureRole.mjTEXROLE_RGB]:
+      # setting mesh uv properties
+      mesh_texcoord, _ = self._get_uv_geometry()
+      self.texcoords = UsdGeom.PrimvarsAPI(self.usd_mesh).CreatePrimvar(
+          "UVMap", Sdf.ValueTypeNames.TexCoord2fArray, UsdGeom.Tokens.faceVarying
+      )
 
-    self.texcoords.Set(mesh_texcoord)
-    self.texcoords.SetIndices(Vt.IntArray(list(range(mesh_facenum * 3))))
+      self.texcoords.Set(mesh_texcoord)
+      self.texcoords.SetIndices(Vt.IntArray(list(range(mesh_facenum * 3))))
 
-    self._set_refinement_properties(self.usd_prim)
-
-    if self.texture_file:
       self.attach_image_material(self.usd_mesh)
     else:
       self.attach_solid_material(self.usd_mesh)
 
   def generate_primitive_mesh(self):
     """Generates the mesh for the primitive USD object."""
-    _, prim_mesh = shapes_component.mesh_generator(self.mesh_config)
+    geom_rgb_texture = self.geom_textures[mujoco.mjtTextureRole.mjTEXROLE_RGB] if self.geom_textures else None
+    texture_type = geom_rgb_texture[1] if geom_rgb_texture else None
+    _, prim_mesh = shapes_module.mesh_factory(self.mesh_config, texture_type)
     prim_mesh.translate(-prim_mesh.get_center())
     return prim_mesh
 
   def _get_uv_geometry(self):
-    assert self.prim_mesh
+    assert self.prim_mesh and self.prim_mesh.triangle_uvs is not None
 
     mesh_texcoord = np.array(self.prim_mesh.triangle_uvs)
     mesh_facetexcoord = np.asarray(self.prim_mesh.triangles)
-    # TODO(etom): bring back support for rescaling the texture coordinates.
+    
+    geom_rgb_texture = self.geom_textures[mujoco.mjtTextureRole.mjTEXROLE_RGB][1]
```

Comment:
Don't you need to handle the case where the texture tuple is None?

Reply:
I believe I make this check prior to calling the function. In the __init__ function, I say:
```
if geom.matid and self.geom_textures[mujoco.mjtTextureRole.mjTEXROLE_RGB]:
      # setting mesh uv properties
      mesh_texcoord, _ = self._get_uv_geometry()
```
The if statement checks that the geom has a material and the RGB texture exists.